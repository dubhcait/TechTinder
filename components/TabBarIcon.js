import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import {Image} from 'react-native';
const homeIcon = require('../assets/images/Asset1.png')
const LinksIcon = require('../assets/images/Asset2.png')


export default function TabBarIcon(props) {
  return (
    <Image
    style={[{width: 40, height: 40, resizeMode: 'contain'}, props.focused ? {opacity: 1} : {opacity : 0.6}]}
    source={props.name == 'home'? homeIcon: LinksIcon } 
  />
    
  );
}
