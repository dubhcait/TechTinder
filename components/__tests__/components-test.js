import React from 'react';
import renderer from 'react-test-renderer';

import { MonoText } from '../StyledText';
import TabBarIcon from '../TabBarIcon';

it(`renders correctly`, () => {
  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();

  expect(tree).toMatchSnapshot();
});

describe('<TabBarIcon /> link', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TabBarIcon name='link' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<TabBarIcon /> home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TabBarIcon name='home' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});