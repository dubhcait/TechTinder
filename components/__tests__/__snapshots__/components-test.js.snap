// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBarIcon /> home renders correctly 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": 40,
        "resizeMode": "contain",
        "width": 40,
      },
      Object {
        "opacity": 0.6,
      },
    ]
  }
/>
`;

exports[`<TabBarIcon /> link renders correctly 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": 40,
        "resizeMode": "contain",
        "width": 40,
      },
      Object {
        "opacity": 0.6,
      },
    ]
  }
/>
`;

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
